src/config.rs

```diff
@@ -82,29 +128,28 @@ impl Config {
 
     fn filter_based_on_rule_argument(
         &self,
-        active_rules: Vec<Arc<dyn Rule>>,
-    ) -> Vec<Arc<dyn Rule>> {
+        active_rules_and_associated_plugins: Vec<(Arc<dyn Rule>, Option<PluginIndex>)>,
+    ) -> Vec<(Arc<dyn Rule>, Option<PluginIndex>)> {
         match self.rule.as_ref() {
             Some(rule_arg) => {
-                let filtered = active_rules
+                let filtered = active_rules_and_associated_plugins
                     .into_iter()
-                    .filter(|rule| &rule.meta().name == rule_arg)
+                    .filter(|(rule, _)| &rule.meta().name == rule_arg)
                     .collect::<Vec<_>>();
                 if !filtered.is_empty() {
                     return filtered;
                 }
                 self.rule_argument_error();
             }
-            None => active_rules,
+            None => active_rules_and_associated_plugins,
         }
     }
 
     fn rule_argument_error(&self) -> ! {
         let rule_arg = self.rule.as_ref().unwrap();
         if self
-            .all_rules
-            .iter()
-            .any(|rule| &rule.meta().name == rule_arg)
+            .get_rules_by_plugin_prefixed_name()
+            .contains_key(rule_arg)
```

Comment:
No tests for this yet but this should make it so that you can pass a "plugin-prefixed" rule name to `--rule` eg `--rule some-plugin/some-rule`